mqtt/mqtt-bridge/src/settings.rs

```diff
@@ -0,0 +1,54 @@
+use config::{Config, ConfigError, Environment};
+use serde::Deserialize;
+
+#[derive(Debug, Clone, Default, PartialEq, Deserialize)]
+pub struct Settings {
+    #[serde(flatten)]
+    nested_bridge: NestedBridgeSettings,
+}
+
+impl Settings {
+    pub fn new() -> Result<Self, ConfigError> {
+        let mut config = Config::new();
+        config.merge(Environment::with_prefix("iotedge"))?;
+
+        config.try_into()
+    }
+
+    pub fn nested_bridge(&self) -> &NestedBridgeSettings {
+        &self.nested_bridge
+    }
+}
+
+#[derive(Debug, Clone, Default, PartialEq, Deserialize)]
+pub struct NestedBridgeSettings {
+    #[serde(rename = "gatewayhostname")]
```

Comment:
nit: try to see if `#[serde(rename_all="lowercase")]` on the struct will work for you. https://serde.rs/container-attrs.html

Reply:
I tried with rename_all and is not finding the env variable, tried with UPPERCASE too and it didn't work either